### PR TITLE
[SPARK-14792][SQL] Move as many parsing rules as possible into SQL parser

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch moves as many parsing rules as possible into SQL parser. There are only three more left after this patch: (1) run native command, (2) analyze, and (3) script IO. These 3 will be dealt with in a follow-up PR.
## How was this patch tested?

No test change. This simply moves code around.

### CODE REVIEW

**Comment by rxin** [comment]
Note that this is based on https://github.com/apache/spark/pull/12550

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala [hunk 1]
```diff
@@ -789,4 +794,242 @@ class SparkSqlAstBuilder extends AstBuilder {
   override def visitConstantList(ctx: ConstantListContext): Seq[String] = withOrigin(ctx) {
     ctx.constant.asScala.map(visitStringConstant)
   }
+
+  /**
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: all the diff this in file beyond this line are just simple copy paste.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 2]
```diff
@@ -41,74 +40,15 @@ import org.apache.spark.sql.catalyst.plans.logical
...
+import org.apache.spark.sql.execution.command.CreateViewAsSelectLogicalCommand
 import org.apache.spark.sql.execution.datasources.{Partition => _, _}
 import org.apache.spark.sql.execution.datasources.parquet.{DefaultSource => ParquetDefaultSource, ParquetRelation}
 import org.apache.spark.sql.hive.client._
 import org.apache.spark.sql.hive.execution.HiveNativeCommand
 import org.apache.spark.sql.hive.orc.{DefaultSource => OrcDefaultSource}
+import org.apache.spark.sql.internal.{HiveSerDe, SQLConf}
 import org.apache.spark.sql.types._
 
-private[hive] case class HiveSerDe(
```
Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: this was moved into sql core

File: sql/core/src/main/scala/org/apache/spark/sql/internal/HiveSerDe.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.internal
+
+case class HiveSerDe(
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/HiveSerDe.scala]: entire file copied directly from hive module

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala [hunk 4]
```diff
@@ -124,32 +124,6 @@ case class AnalyzeTable(tableName: String) extends RunnableCommand {
 }
 
 private[hive]
-case class AddJar(path: String) extends RunnableCommand {
```
Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala]: moved into sql core resources.scala

**Comment by rxin** [comment]
cc @hvanhovell

**Comment by yhuai** [comment]
LGTM

**Comment by hvanhovell** [comment]
LGTM

**Comment by rxin** [comment]
I'm going to merge this and push a commit to disable the failing tests. It looks like they have been flaky from an earlier pr.

**Pull request closed without merging**